src/api/app/controllers/webui/badge_controller.rb

```diff
@@ -0,0 +1,149 @@
+class Webui::BadgeController < Webui::WebuiController
+  before_action :set_project, only: [:show]
+  before_action :require_package, only: [:show]
+  before_action :require_login, except: [:show]
+
+  class Badge
```

Comment:
Can you move the class to `app/models/` directory, please? The convention is to have the objects there.